packages/data/src/test/registry.js

```diff
@@ -358,6 +358,150 @@ describe( 'createRegistry', () => {
+				},
+				resolvers: {
+					getPage: fulfilledResolver,
+				},
+			} );
+
+			const promise1 = registry.resolveSelect( 'demo' ).getPage( 1 );
+			jest.runAllTimers();
+			const result1 = await promise1;
+			expect( result1 ).toEqual( {
+				title: 'Page 1',
+				content: 'Content 1',
+			} );
+
+			const promise2 = registry.resolveSelect( 'demo' ).getPage( 2 );
+			jest.runAllTimers();
+			const result2 = await promise2;
+			expect( result2 ).toEqual( {
+				title: 'Page 2',
+				content: 'Content 2',
+			} );
+
+			// Resolver should not be called since isFulfilled returns truthy
+			expect( fulfilledResolver ).not.toHaveBeenCalled();
+		} );
+
+		it( 'should call resolver when isFulfilled returns false', async () => {
+			const fulfill = jest.fn().mockImplementation( () => ( {
+				type: 'SET_DATA',
+				data: 'resolved data',
+			} ) );
+			const isFulfilled = jest.fn( ( state ) => state.hasData );
+
+			registry.registerStore( 'demo', {
+				reducer: ( state = { hasData: false }, action ) => {
+					if ( action.type === 'SET_DATA' ) {
+						return { hasData: true, data: action.data };
+					}
+					return state;
+				},
+				selectors: {
+					getData: ( state ) => state.data,
+				},
+				resolvers: {
+					getData: { fulfill, isFulfilled },
+				},
+			} );
+
+			const promise = registry.resolveSelect( 'demo' ).getData();
+			jest.runAllTimers();
+			const result = await promise;
+
+			// Initial state has hasData: false, so resolver should be called
+			expect( isFulfilled ).toHaveBeenCalledTimes( 1 );
+			expect( fulfill ).toHaveBeenCalledTimes( 1 );
+			expect( result ).toBe( 'resolved data' );
+
+			// Subsequent call should use cached result, not calling isFulfilled or fulfill again
+			const promise2 = registry.resolveSelect( 'demo' ).getData();
+			jest.runAllTimers();
+			const result2 = await promise2;
+			expect( result2 ).toBe( 'resolved data' );
+			// isFulfilled is only called once since resolution is already marked as finished
+			expect( isFulfilled ).toHaveBeenCalledTimes( 1 );
+			expect( fulfill ).toHaveBeenCalledTimes( 1 ); // Still only called once
+		} );
+
+		it( 'should mark resolution as failed when isFulfilled throws an error', async () => {
+			const fulfill = jest.fn().mockImplementation( () => ( {
+				type: 'SET_DATA',
+				data: 'resolved data',
+			} ) );
+			const isFulfilled = jest.fn( () => {
+				throw new Error( 'isFulfilled error' );
+			} );
+
+			registry.registerStore( 'demo', {
+				reducer: ( state = { hasData: false }, action ) => {
+					if ( action.type === 'SET_DATA' ) {
+						return { hasData: true, data: action.data };
+					}
+					return state;
+				},
+				selectors: {
+					getData: ( state ) => state.data,
+				},
+				resolvers: {
+					getData: { fulfill, isFulfilled },
+				},
+			} );
+
+			const promise = registry.resolveSelect( 'demo' ).getData();
+			jest.runAllTimers();
+			await expect( promise ).rejects.toThrow( 'isFulfilled error' );
```

Comment:
@jsnajdr this is odd. The error is caught and the state of the resolver transitions to failed, but Jest fails the test without this line. I'm still researching the issue.

Reply:
Yes, `resolveSelect` detects the error and returns it.

If `isFulfilled` throws, that is certainly suprising and unexpected and the error should be surfaced. So yes, the resolution should fail.